doc/rst/technotes/partialInstantiations.rst

```diff
@@ -0,0 +1,250 @@
+.. _readme-partial-instantiations:
+
+.. default-domain:: chpl
+
+======================
+Partial Instantiations
+======================
+
+Overview
+--------
+
+A fully generic type expression involving a class or record is one in which
```

Comment:
I'd probably rewrite the first couple sentences. What about this:

A generic class or record can be fully instantiated, fully generic,  or partially instantiated.
A fully instantiated type expression has concrete types for all generic fields.
A fully generic type expression involving a class or record is one in which none of the generic fields of the type have been instantiated.
A partially generic type expression is one in which one or more, but not all, fields have been instantiated.

Note the zero or more -> one or more change. Partially generic needs something to be instantiated right (otherwise it is fully generic?)